token/program/src/processor.rs

```diff
@@ -1674,84 +1670,6 @@ mod tests {
         )
         .unwrap();
 
-        // transfer to self
```

Comment:
Seems like we might want to keep a transfer-to-self test to ensure that a properly constructed and signed one will succeed...?

Reply:
Yep. Realized I got a little carried away with the delete key :sweat_smile: 

Adding it back now